contracts/community-nft1.0.sol

```diff
@@ -0,0 +1,297 @@
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "../utils/StringUtils.sol";
+import "../SponsorWhitelistControl.sol";
+import "../owner/Operator.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+contract TspaceAllnighter is ERC1155, Operator {
+    event TspaceMint(address _onwer, uint256 _tokenId, string metedate);
+
+    using Counters for Counters.Counter;
+    //tokenid自增一
+    Counters.Counter private _tokenIds;
+
+    using SafeMath for uint256;
+
+    // Contract name
+    string public name = "Tspace Allnighter";
+    // Contract symbol
+    string public symbol = "allnighter";
+
+    // Mapping from owner to list of owned token IDs
+    mapping(address => uint256[]) private _ownedTokens;
+
+    // Mapping from token ID to index of the owner tokens list
+    mapping(uint256 => uint256) private _ownedTokensIndex;
+
+    // Array with all token ids, used for enumeration
+    uint256[] private _allTokens;
+
+    // Mapping from token id to position in the allTokens array
+    mapping(uint256 => uint256) private _allTokensIndex;
+
+    string baseUri;
+
+    mapping(uint256 => address[]) public ownerOfAddress;
+
+    //tokenId => metadate
+    mapping(uint256 => string) public tokenMetaDate;
```

Comment:
Per EIP-1155, you could use a common URI with placeholder for all tokens, which also reduce the gas/storage cost.